### PR TITLE
Add Arch-specific libc test case.

### PR DESCRIPTION
**Pull request by XinStellaris**
## Summary
This a testcase for https://github.com/apache/nuttx/pull/8814
## Impact
N/A
## Testing
Tested on QEMU with lm3s6965-ek:qemu-flat

### CODE REVIEW

File: testing/arch_libc/arch_libc_test_main.c [hunk 1]
```diff
@@ -0,0 +1,255 @@
...
+  memset(g_test_dst_str, 0, sizeof(g_test_dst_str));
+
+  for (off = 0; off < TEST_SHORT_SRC_LEN; off++)
+    {
+      g_test_src_str[off + src_offset] = '0' + off % 10;
+    }
+
+  g_test_src_str[off + src_offset] = 0;
+
+  return;
```
Comment (pkarashchenko) [inline testing/arch_libc/arch_libc_test_main.c:81]:
```suggestion
```

File: testing/arch_libc/arch_libc_test_main.c [hunk 2]
```diff
@@ -0,0 +1,255 @@
...
+        {
+          pass = false;
+          printf("src data error, index %d\n", off);
+        }
+    }
+
+  for (off = TEST_SHORT_SRC_LEN; off < TEST_MAX_STRING_LEN - dst_offset;
+       off++)
+    {
+      if (result_str[off] != 0)
```
Comment (pkarashchenko) [inline testing/arch_libc/arch_libc_test_main.c:121]:
```suggestion
      if (result_str[off] != '\0')
```

File: testing/arch_libc/arch_libc_test_main.c [hunk 3]
```diff
@@ -0,0 +1,255 @@
...
+        {
+          pass = false;
+          printf("dest tailing zero error, index %d\n", off);
+        }
+    }
+
+  for (off = TEST_SHORT_SRC_LEN; off < TEST_MAX_STRING_LEN - src_offset;
+       off++)
+    {
+      if (src[off] != 0)
```
Comment (pkarashchenko) [inline testing/arch_libc/arch_libc_test_main.c:131]:
```suggestion
      if (src[off] != '\0')
```

File: testing/arch_libc/arch_libc_test_main.c [hunk 4]
```diff
@@ -0,0 +1,255 @@
...
+  /* strcpy shouln't change arch_libc_test_strcpy's local variable */
+
+  if (dest != (g_test_dst_str + dst_offset) ||
+      src != (g_test_src_str + src_offset))
+    {
+      pass = false;
+      printf("local var changed after calling strcpy\n");
+    }
+
+  if (pass == false)
```
Comment (pkarashchenko) [inline testing/arch_libc/arch_libc_test_main.c:147]:
```suggestion
  if (!pass)
```

File: testing/arch_libc/arch_libc_test_main.c [hunk 5]
```diff
@@ -0,0 +1,251 @@
...
+/****************************************************************************
+ * Pre-processor Definitions
+ ****************************************************************************/
+
+/* Configuration ************************************************************/
+
+#ifdef CONFIG_TESTING_ARCH_LIBC_STRCPY
+#define TEST_MAX_STRING_LEN    128
+#define TEST_SHORT_SRC_LEN     50
+#define TEST_LONG_SRC_LEN      128
```
Comment (pkarashchenko) [inline testing/arch_libc/arch_libc_test_main.c:45]:
```suggestion
#  define TEST_MAX_STRING_LEN    128
#  define TEST_SHORT_SRC_LEN     50
#  define TEST_LONG_SRC_LEN      128
```

**Review by pkarashchenko — APPROVED** [review]
LGTM!

File: testing/arch_libc/arch_libc_test_main.c [hunk 6]
```diff
@@ -0,0 +1,251 @@
...
+ * Name: arch_libc_strcpy_speed
+ ****************************************************************************/
+
+int arch_libc_strcpy_speed(void)
+{
+  int dest_off;
+  int src_off;
+  uint32_t cycles = 0;
+
+  up_perf_init(NULL);
```
Comment (xiaoxiang781216) [inline testing/arch_libc/arch_libc_test_main.c:217]: up_perf_init should be called by board file, not application.

File: testing/arch_libc/Make.defs [hunk 7]
```diff
@@ -0,0 +1,23 @@
+############################################################################
+# apps/testing/mtd_config_fs/Make.defs
```
Comment (xiaoxiang781216) [inline testing/arch_libc/Make.defs:2]:
```suggestion
# apps/testing/arch_libc/Make.defs
```

File: testing/arch_libc/Kconfig [hunk 8]
```diff
@@ -0,0 +1,38 @@
...
+		This is the name of the program that will be used when the NSH ELF
+		program is installed.
+
+config TESTING_ARCH_LIBC_PRIORITY
+	int "arch libc test priority"
+	default 100
+
+config TESTING_ARCH_LIBC_STACKSIZE
+	int "arch libc test stack size"
+	default 4096
```
Comment (xiaoxiang781216) [inline testing/arch_libc/Kconfig:32]: why not use the default stack size

**Review by acassis — APPROVED** [review]

**Pull request merged**